agent/lib/kontena/workers/volumes/volume_manager.rb

```diff
@@ -68,15 +71,15 @@ def populate_volumes_from_docker
     def ensure_volume(volume)
       debug "ensuring volume existence: #{volume.inspect}"
       begin
-        Docker::Volume.get(volume.name)
-      rescue Docker::Error::NotFoundError
-        info "creating volume"
-        v = Docker::Volume.create(volume.name, {
-          'Driver' => volume.driver,
-          'DriverOpts' => volume.driver_opts,
-          'Labels' => volume.labels
-        })
-        sync_volume_to_master(v)
+        unless volume_exist?(volume.name, volume.driver)
+          info "creating volume"
+          v = Docker::Volume.create(volume.name, {
+            'Driver' => volume.driver,
+            'DriverOpts' => volume.driver_opts,
+            'Labels' => volume.labels
+          })
+          sync_volume_to_master(v)
+        end
       rescue => exc
```

Comment:
I'm guessing that this wasn't rescuing `Docker::Volume.create` errors from the earlier `rescue Docker::Error::NotFoundError` before, leading to actor crashes?